crates/rattler_conda_types/src/match_spec/mod.rs

```diff
@@ -89,7 +89,7 @@ use matcher::StringMatcher;
 /// assert_eq!(spec.version, Some(VersionSpec::from_str("1.0.*", Strict).unwrap()));
 /// assert_eq!(spec.channel, Some(Channel::from_str("conda-forge", &channel_config).map(|channel| Arc::new(channel)).unwrap()));
 ///
-/// let spec = MatchSpec::from_str("conda-forge/linux-64::foo >=1.0", Strict).unwrap();
+/// let spec = MatchSpec::from_str(r#"conda-forge::foo >=1.0[subdir="linux-64"]"#, Strict).unwrap();
```

Comment:
Can this be changed back with the new logic? Or is this still invalid?

Reply:
Did you one better:
https://github.com/conda-incubator/rattler/commit/36a03b3500cde36aa292160a2b8277ea8d5b9d82